common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp

```diff
@@ -409,6 +461,95 @@ bool mon_wlan_hal_dwpal::sta_link_measurements_11k_request(const std::string &st
+        return false;
+    }
+
+    // prepare new scan params with changed dwell time
+    memcpy_s(&new_fg, sizeof(new_fg), &org_fg, sizeof(org_fg));
+    memcpy_s(&new_bg, sizeof(new_bg), &org_bg, sizeof(org_bg));
+    new_fg.passive = dwell_time_msec;
+    new_fg.active  = dwell_time_msec;
+    new_bg.passive = dwell_time_msec;
+    new_bg.active  = dwell_time_msec;
+
+    // set new scan params & get newly set values for validation
+    if (!dwpal_set_scan_params_fg(new_fg) || !dwpal_set_scan_params_bg(new_bg)) {
+        LOG(ERROR) << "Failed setting new values, restoring original scan parameters";
+        dwpal_set_scan_params_fg(org_fg);
+        dwpal_set_scan_params_bg(org_bg);
+        return false;
+    }
+    if (!dwpal_get_scan_params_fg(new_fg) || !dwpal_get_scan_params_bg(new_bg) ||
+        (new_fg.active != dwell_time_msec) || (new_fg.passive != dwell_time_msec) ||
+        (new_bg.active != dwell_time_msec) || (new_bg.passive != dwell_time_msec)) {
+        LOG(ERROR) << "Validation failed, restoring original scan parameters";
+        dwpal_set_scan_params_fg(org_fg);
+        dwpal_set_scan_params_bg(org_bg);
+        return false;
+    }
+
+    // get frequancies from channel pool and set in scan_params
+    if (!dwpal_get_freq(channel_pool, m_radio_info.channel, m_radio_info.iface_name, channel_scan_params)) {
+        LOG(ERROR) << "Failed getting frequencies, restoring original scan parameters";
+        dwpal_set_scan_params_fg(org_fg);
+        dwpal_set_scan_params_bg(org_bg);
+        return false;
+    }
+    LOG(DEBUG) << __func__ << " frequancies set in scan_params";
+    // must as single wifi won't allow scan on ap without this flag
+    channel_scan_params.ap_force = 1;
+
+    if (dwpal_driver_nl_scan_trigger(get_dwpal_nl_ctx(), (char *)m_radio_info.iface_name.c_str(),
+                                     &channel_scan_params) != DWPAL_SUCCESS) {
+        LOG(ERROR) << " scan trigger failed! Abort scan, restoring original scan parameters";
+        dwpal_set_scan_params_fg(org_fg);
+        dwpal_set_scan_params_bg(org_bg);
+        return false;
+    }
+
+    // timeout for nl confirmation on scan trigger
+    std::this_thread::sleep_for(std::chrono::milliseconds(1));
+
+    // restore scan params with original dwell time
+    // no reason to check since we restore the original params here anyway
+    // and the next validation will validate the change
+    dwpal_set_scan_params_fg(org_fg);
+    dwpal_set_scan_params_bg(org_bg);
+
+    // validate if "set" function to original values worked
+    if (!dwpal_get_scan_params_fg(new_fg) || !dwpal_get_scan_params_bg(new_bg) ||
+        (new_fg.active != org_fg.active) || (new_fg.passive != org_fg.passive) ||
+        (new_bg.active != org_bg.active) || (new_bg.passive != org_bg.passive)) {
+        LOG(ERROR) << "Validation failed, original scan parameters were not restored";
+        return false;
+    }
```

Comment:
Why is it required to restore the original values?

Reply:
currently channel scan API  doesn't include dwell time parameter so we are changing driver operational scan parameters on the fly.
added comment in code.